doc/user/load_save.md

```diff
@@ -1,61 +1,563 @@
+   operators and members.
+ * Update the formats table in #formats.
+ * Add a section with examples about the type of data to be loaded.
+ * Where relevant, add links to the new data type or new options class
+   throughout.
+
+Then go read it and make sure it all makes sense... :)
+-->
+
 # Data loading and I/O
 
 mlpack provides the `data::Load()` and `data::Save()` functions to load and save
 [Armadillo matrices](matrices.md) (e.g. numeric and categorical datasets) and
 any mlpack object via the [cereal](https://uscilab.github.io/cereal/)
 serialization toolkit.  A number of other utilities related to loading and
-saving data and objects are also available.
-
- * [Numeric data](#numeric-data)
- * [Mixed categorical data](#mixed-categorical-data)
-   - [`data::DatasetInfo`](#datadatasetinfo)
-   - [Loading categorical data](#loading-categorical-data)
- * [Image data](#image-data)
-   - [`data::ImageInfo`](#dataimageinfo)
-   - [Loading images](#loading-images)
- * [mlpack objects](#mlpack-objects): load or save any mlpack object
- * [Formats](#formats): supported formats for each load/save variant
+saving data and objects are also available.  The `data::Load()` and
+`data::Save()` functions have numerous options to configure load/save behavior
+and [format detection/selection](#formats).
+
+## `data::Load()`
+
+ - `data::Load(filename, X)`
+ - `data::Load(filename, X, Option1 + Option2 + ...)`
+ - `data::Load(filename, X, opts)`
+   * Load `X` from a single file `filename` (of type `std::string`) with
+     default options:
+     - the format of the file is [auto-detected](#formats) based on the
+       extension of the file, and
+     - an exception is *not* thrown on an error.
+   * `X` can be [any supported load type](#types).
+   * The given options `Options1, Options2 ...` must be from the
+     [list of standalone operators](#dataoptions) and be appropriate for the type
+     of `X`.
+   * `opts` is a [`DataOptions` object](#dataoptions) whose subtype matches the
+     type of `X`.
+   * Returns a `bool` indicating whether the load was a success.
+
+For some types of data, it is also possible to load multiple images at once from a set of files:
+
+ - `data::Load(filenames, X)`
+ - `data::Load(filenames, X, Option1 + Option2 + ...)`
+ - `data::Load(filenames, X, opts)`
+    * Load data from `filenames` (a `std::vector<std::string>`) into the matrix `X`.
+    - For [numeric data](#numeric-data), data loaded from each file is concatenated into `X`.
+    - For [image data](#image-data), each image is flattened into one column of `X`.
+    - Metadata (e.g. image size, number of columns, etc.) in all files in `filenames` must match or loading will fail.
+    - Loading options can be specified by either standalone options or an instantiated [`DataOptions` object](#dataoptions).
+    - an exception is *not* thrown on an error.
+   * `X` can be [any supported load type](#types).
```

Comment:
Not true, it can only be numeric data or image data.  We should remove this bullet point.